vignettes/dp-variance.Rmd

```diff
@@ -0,0 +1,148 @@
+data <- data.frame(x1, x2)
+
+# example on a numeric variable
+dpVarianceExample <- dpVariance$new(mechanism='mechanismLaplace', varType='numeric', 
+                      variable='x1', epsilon=10, n=10, rng=c(0, 70))
+dpVarianceExample$release(data)
+print(dpVarianceExample$result)
+
+# example on a logical variable
+dpVarianceExample2 <- dpMean$new(mechanism='mechanismLaplace', varType='logical', 
+                       variable='x2', epsilon=0.1, n=10, rng=c(0, 1))
+dpVarianceExample2$release(data)
+print(dpVarianceExample2$result)
+```
+
+Arguments
+---------
+
+In typical usage, there are two methods to the `dpVariance` class: the `new` method and the `release` method. The `new` method creates an object of the class and accepts the following arguments:
+
+* `mechanism` \ Character, the class name of the mechanism used to perturb the true estimate, must be `'mechanismLaplace'`.
+
+* `vaType` \ Character, the type of values in the data frame that will be passed to the mechanism. Should be one of `'numeric'`, `'integer'`, or `'logical'`.
+
+* `variable` \ character, the name of the variable in the data for which to calculate the variance.
+
+* `n` \ Integer, the number of observations in the vector.
+
+* `rng` \ Numeric, a 2-tuple giving an a priori estimate of the lower and upper bounds of the vector.
+
+* `epsilon` \ Numeric, the differential privacy parameter $\epsilon$, typically taking values between 0 and 1 and reflecting the privacy cost of the query. Optional, default `NULL`. If `NULL`, the user must specify a value for `accuracy`.
+
+* `accuracy` \ Numeric, the accuracy of the query. Optional, default `NULL`. If `NULL`, the user must specify a value for `epsilon`. If `epsilon` is not `NULL`, this value is ignored and evaluated internally.
+
+* `imputeRng` \ Numeric, a 2-tuple giving a range within which missing values of the vector are imputed. Optional, default `NULL`. If `NULL`, missing values are imputed using the range provided in `rng`. See Notes below for more information.
+
+* `alpha` \ Numeric, the statistical significance level used in evaluating accuracy and privacy parameters. If the bootstrap is employed, `alpha` is also used to trim the release. Default `0.05`.
+
+&nbsp; 
+
+The `release` method calculates the mean of given data and accepts a single argument:
+
+* `data` \ Data frame containing numeric columns corresponding the name specified in `variable`.
+
+Examples
+------
+
+Import the PSIlence library and attach the sample dataset:
+
+```{r, eval = TRUE}
+library(PSIlence)
+data(PUMS5extract10000)
+```
+
+&nbsp;
+
+__Numeric Example__
+
+To calculate a private variance of a numeric vector with `dpVariance`, enter the mechanism (this will be the Laplace Mechanism, or 'mechanismLaplace'), the variable type ('numeric'), the variable of interest (the column name of the variable in the dataframe), the number of observations in the dataframe, the epsilon value (generally less than 1), and the range:
```

Comment:
Maybe add backticks? `mechanismLaplace`, `numeric`.

Reply:
I did not do that because these are not code snippets, they are entered as strings